packages/admin-next/dashboard/src/v2-routes/locations/location-create/components/create-location-form/create-location-form.tsx

```diff
@@ -0,0 +1,235 @@
+import { zodResolver } from "@hookform/resolvers/zod"
+import { Button, Heading, Input, Text } from "@medusajs/ui"
+import { useAdminCreateStockLocation } from "medusa-react"
+import { useForm } from "react-hook-form"
+import { useTranslation } from "react-i18next"
+import * as zod from "zod"
+import { CountrySelect } from "../../../../../components/common/country-select"
+import { Form } from "../../../../../components/common/form"
+import { RouteFocusModal } from "../../../../../components/route-modal"
+
+const CreateLocationSchema = zod.object({
+  name: zod.string().min(1),
+  address: zod.object({
+    address_1: zod.string().min(1),
+    address_2: zod.string().optional(),
+    country_code: zod.string().min(2).max(2),
+    city: zod.string().optional(),
+    postal_code: zod.string().optional(),
```

Comment:
q: shouldn't this be required?

Reply:
it's not required in the api or the db, then we should align what we require and what we don't across frontend and backend 😄